src/InstrumentationEngine.Attach/Program.cs

```diff
@@ -0,0 +1,170 @@
+
+        internal static void Main(string[] args)
+        {
+            // TODO: Parse command line for additional arguments such as configuration files
+            if (args.Length != 1)
+            {
+                WriteErrorJson("Expected single argument that represents the process ID to which the engine shall be attached.");
+                return;
+            }
+
+            string processIdString = args[0];
+
+            if (!Int32.TryParse(processIdString, out int processId))
+            {
+                WriteErrorJson(Invariant($"Could not parse process ID argument '{processIdString}' to integer."));
+                return;
+            }
+
+            Process process;
+            try
+            {
+                process = Process.GetProcessById(processId);
+            }
+            catch (Exception ex)
+            {
+                WriteErrorJson(ex.Message);
+                return;
+            }
+
+            // This executable should be in the [Root]\Tools\Attach directory. Get the root directory
+            // and build back to the instrumentation engine file path.
+
+            // CONSIDER: Is there a better way to get the engine path instead of building the path
+            // relative to the current executable? For example, requiring the root path of the engine
+            // to be passed as a parameter (which increases the difficulty of using this executable).
+
+            string? attachDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            if (null == attachDirectory || !Directory.Exists(attachDirectory))
+            {
+                WriteErrorJson(Invariant($"Directory '{attachDirectory}' does not exist."));
+                return;
+            }
+
+            string? toolsDirectory = Path.GetDirectoryName(attachDirectory);
+            if (null == toolsDirectory || !Directory.Exists(toolsDirectory))
+            {
+                WriteErrorJson(Invariant($"Directory '{toolsDirectory}' does not exist."));
+                return;
+            }
+
+            string? rootDirectory = Path.GetDirectoryName(toolsDirectory);
+            if (null == rootDirectory || !Directory.Exists(rootDirectory))
+            {
+                WriteErrorJson(Invariant($"Directory '{rootDirectory}' does not exist."));
+                return;
+            }
+
+            string enginePath;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                enginePath = Path.Combine(rootDirectory, "Instrumentation64", "libInstrumentationEngine.so");
+            }
+            else
+            {
+                bool isWow64;
+                try
+                {
+                    // isWow64 is true for 32-bit applications running on a 64-bit OS
+                    if (!NativeMethods.IsWow64Process(process.Handle, out isWow64))
```

Comment:
>NativeMethods.IsWow64Process [](start = 25, length = 28)

Amy solution for Linux? Or do we only support 64bit on Linux?

Reply:
We currently only support 64bit Linux. I don't think we have plans to support 32bit.

---
In reply to: [384714445](https://github.com/microsoft/CLRInstrumentationEngine/pull/212#discussion_r384714445) [](ancestors = 384714445)

Reply:
On Linux, there is no 32 bit support for .NET Core (except for ARM, but CLRIE does not support that at all).

---
In reply to: [384714445](https://github.com/microsoft/CLRInstrumentationEngine/pull/212#discussion_r384714445) [](ancestors = 384714445)